src/Raven.Server/Documents/Handlers/Debugging/QueriesDebugHandler.cs

```diff
@@ -93,5 +95,112 @@ public Task RunningQueries()
 
             return Task.CompletedTask;
         }
+       
+        [RavenAction("/databases/*/debug/queries/cache/list", "GET", AuthorizationStatus.ValidUser, IsDebugInformationEndpoint = true)]
+        public Task QueriesCacheList()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var queryCache = Database.QueryMetadataCache.GetQueryCache();
+
+                var responseDjv = new DynamicJsonValue();
+                var queriesList = new DynamicJsonArray();
+
+                responseDjv["CachedQueryMetadataItems"] = queriesList;
```

Comment:
We are using "Results" most often. Let's stay consistent.

Reply:
IIRC writer has a method for that `WriteArray`

Reply:
nice